docs/Troubleshooting/resolving-splitbrain.md

```diff
@@ -308,15 +308,312 @@ Healed gfid:b23dd8de-af03-4006-a803-96d8bc0df004.
+
+Execute the following command:
+```console
+# gluster volume heal VOLNAME split-brain source-brick HOSTNAME:export-directory-absolute-path FILE
+```
+
+In this command, FILE present in HOSTNAME : export-directory-absolute-path is taken as source for healing.
+
+### Example :
+```console
+# gluster volume heal testvol split-brain source-brick 10.70.47.144:/bricks/brick2/b1 /f3
+GFID split-brain resolved for file /f3
+```
+
+After the healing is complete, the GFID of the file on both the bricks should be same as that of the brick which was chosen as source for healing. The following is a sample output of the getfattr command after the file is healed.
+
+On brick /b0
+```console
+# getfattr -d -m . -e hex /bricks/brick2/b0/f3
+getfattr: Removing leading '/' from absolute path names
+file: bricks/brick2/b0/f3
+security.selinux=0x73797374656d5f753a6f626a6563745f723a676c7573746572645f627269636b5f743a733000
+trusted.gfid=0x90d9b0f65f6530b95b9f3f8334033df
+trusted.gfid2path.364f55367c7bd6f4=0x30303030303030302d303030302d303030302d303030302d3030303030303030303030312f6634
+```
+
+On brick /b1
+```console
+# getfattr -d -m . -e hex /bricks/brick2/b1/f3
+getfattr: Removing leading '/' from absolute path names
+file: bricks/brick2/b1/f3
+security.selinux=0x73797374656d5f753a6f626a6563745f723a676c7573746572645f627269636b5f743a733000
+trusted.gfid=0x90d9b0f65f6530b95b9f3f8334033df
+trusted.gfid2path.364f55367c7bd6f4=0x30303030303030302d303030302d303030302d303030302d3030303030303030303030312f6634
+```
+
+> **Note**
+>- One can not use the GFID of the file as an argument with any of the CLI options to resolve GFID split-brain. It should be the absolute path as seen from the mount point to the file considered as source.
+>
+>- With source-brick option there is no way to resolve all the GFID split-brain in one shot by not specifying any file path in the CLI as done while resolving data or metadata split-brain. For each file in GFID split-brain, run the CLI with the policy you want to use.
+>
+>- Resolving directory GFID split-brain using CLI with the "source-brick" option in a "distributed-replicated" volume needs to be done on all the sub-volumes explicitly, which are in this state. Since directories get created on all the sub-volumes, using one particular brick as source for directory GFID split-brain heals the directory for that particular sub-volume. Source brick should be chosen in such a way that after heal all the bricks of all the sub-volumes have the same GFID.
+
 ## Note:
-As mentioned earlier, entry split-brain and gfid split-brain healing
- are not supported using CLI. Trying to heal /dir would fail as it is in entry split-brain.
+As mentioned earlier, entry/type-mismatch split-brain can not be resolved using CLI. Trying to heal such entry would fail.
 
 ### Example
+The entry named "entry1" is of different types on the bricks of the replica. Lets try to heal that using the split-brain CLI.
```

Comment:
Do we explain what type-mismatch is somewhere in the doc? If we don't, it is better to explicitly state that it  means different `st_mode` values (i.e. the entry is a file in one brick while it is a directory on the other). Otherwise the meaning of 'different type' might not be apparent to users.

Reply:
Added this to the previous line.